src/go/configgenerator/filtergen/jwt_authn.go

```diff
@@ -39,52 +39,92 @@ const (
 )
 
 type JwtAuthnGenerator struct {
-	// skipFilter indicates if this filter is disabled based on options and config.
-	skipFilter bool
+	// ServiceName is the service config name.
+	ServiceName string
+
+	// AuthConfig is the full authentication config from the OP service config.
+	// TODO(nareddyt): This should be generalized, but since we only use jwt_authn
+	// filter in ESPv2, we can generalize it later.
+	AuthConfig *confpb.Authentication
+
+	// AuthRequiredBySelector maps which selectors require per-route level authn
+	// config.
+	AuthRequiredBySelector map[string]bool
+
+	// General options below.
+
+	HttpRequestTimeout    time.Duration
+	GeneratedHeaderPrefix string
+
+	// JWT Authn specific options below.
+
+	JwksCacheDurationInS               int
+	DisableJwksAsyncFetch              bool
+	JwksAsyncFetchFastListener         bool
+	JwksFetchNumRetries                int
+	JwksFetchRetryBackOffBaseInterval  time.Duration
+	JwksFetchRetryBackOffMaxInterval   time.Duration
+	JwtPadForwardPayloadHeader         bool
+	DisableJwtAudienceServiceNameCheck bool
```

Comment:
Make these private if they don't have to be public.

Reply:
See comment above